vinyldns/resource_group.go

```diff
@@ -144,20 +161,42 @@ func userSchema() *schema.Schema {
 
 func users(userType string, d *schema.ResourceData) []vinyldns.User {
 	users := []vinyldns.User{}
-	usersCount := d.Get(fmt.Sprintf("%s.#", userType)).(int)
-
-	for i := 0; i < usersCount; i++ {
-		prefix := fmt.Sprintf("%s.%d", userType, i)
-
-		users = append(users, vinyldns.User{
-			UserName:  d.Get(prefix + ".user_name").(string),
-			FirstName: d.Get(prefix + ".first_name").(string),
-			LastName:  d.Get(prefix + ".last_name").(string),
-			Email:     d.Get(prefix + ".email").(string),
-			Created:   d.Get(prefix + ".created").(string),
-			ID:        d.Get(prefix + ".id").(string),
-		})
+
+	if u, ok := d.GetOk(userType); ok {
+		schemaUsers := u.(*schema.Set).List()
+
+		for _, user := range schemaUsers {
+			u := user.(map[string]interface{})
+
+			users = append(users, vinyldns.User{
+				UserName:  u["user_name"].(string),
+				FirstName: u["first_name"].(string),
+				LastName:  u["last_name"].(string),
+				Email:     u["email"].(string),
+				Created:   u["created"].(string),
+				ID:        u["id"].(string),
+			})
+		}
 	}
 
 	return users
 }
+
+func schemaUsers(users []vinyldns.User) []map[string]interface{} {
```

Comment:
This naming strikes me as a bit confusing. We already have `userSchema()` and this is `schemaUsers()`